torch/csrc/jit/script/init.cpp

```diff
@@ -702,6 +720,34 @@ void initJitScriptBindings(PyObject* module) {
         didFinishEmitFunction(ret);
         return ret;
       });
+  m.def(
+      "_jit_script_compile_overload",
+      [](const std::string& qualname,
+         const Decl& overload_decl,
+         const Def& implementation_def,
+         ResolutionCallback rcb,
+         const FunctionDefaults& defaults) {
+        const auto name = c10::QualifiedName(qualname);
+        auto cu = get_python_cu();
+        checkOverloadDecl(overload_decl, implementation_def.decl());
+        auto new_def = implementation_def.withDecl(overload_decl);
```

Comment:
It looks like replacing the `implementation_def` is all this does differently from `_jit_script_compile`, that should be moved to a function so we don't have to copy paste it here